include/aspect/world_builder/interface.h

```diff
@@ -0,0 +1,269 @@
+         * the SimulatorAccess (if applicable) is initialized.
+         */
+        virtual void initialize ();
+
+        /**
+         * Return the temperature of an object.
+         */
+        virtual
+        void temperature (const dealii::Point<dim> &position,
+                          const unsigned int object_number,
+                          const WorldBuilderParameters<dim> &world_builder_parameters,
+                          double &temperature) const = 0;
+
+        /**
+         * Return the composition of an object.
+         */
+        virtual
+        void composition (const dealii::Point<dim> &position,
+                          const unsigned int n_comp, const unsigned int object_number,
+                          const WorldBuilderParameters<dim> &world_builder_parameters,
+                          double &composition) const = 0;
+
+        /**
+         * Declare the parameters this class takes through input files. The
+         * default implementation of this function does not describe any
+         * parameters. Consequently, derived classes do not have to overload
+         * this function if they do not take any runtime parameters.
+         */
+        static
+        void
+        declare_parameters (WorldBuilderParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         * The default implementation of this function does not read any
+         * parameters. Consequently, derived classes do not have to overload
+         * this function if they do not take any runtime parameters.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+        //static WorldBuilderParameterHandler* ph;
+
+        /*static
+          void
+          register_world_builder(ParameterHandler &prm);*/
+
+      private:
+
+    };
+
+
+
+
+
+    /**
+     * Manager class
+     */
+    template <int dim>
+    class Manager
+    {
+      public:
+        ParameterHandler wb_prm;
+        static WorldBuilderParameterHandler ph;
+        std::map<std::string,Interface<dim>* >  modules;
+
+    };
+
+
+    /**
+     * Register a initial topography model so that it can be selected from the parameter
+     * file.
+     *
+     * @param name A string that identifies the initial topography model
+     * @param description A text description of what this model does and that
+     * will be listed in the documentation of the parameter file.
+     * @param declare_parameters_function A pointer to a function that can be
+     * used to declare the parameters that this initial topography model wants
+     * to read from input files.
+     * @param factory_function A pointer to a function that can create an
+     * object of this initial topography model.
+     *
+     * @ingroup WorldBuilder
+     */
+    template <int dim>
+    void
+    register_world_builder (const std::string &name,
```

Comment:
This function and the `create_world_builder` function are not the best way to implement this. I suppose you took this from one of the other plugin systems without manager, but it is simpler to use the same structure as for the `InitialTemperature::Manager` class. All the stuff in `create_world_builder` can either move into the constructor of `Manager` (if it is related to the core of `Manager`), or into `parse_parameters` (if it is related to the plugins). `register_world_builder` can become a member function of `Manager`, and `declare_parameters` can also become member functions of `Manager`.